File: problems/src/test/java/StringToIntegerAtoiTest.java
import org.junit.Test;
import testing.Sample;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class StringToIntegerAtoiTest extends SampledTest {
    @Override
    public String getFileName() {
        return "string_to_integer_atoi.json";
    }

    @Test
    public void test() {
        for (TypedSample sample : getSamples(TypedSample[].class)) {
            for (StringToIntegerAtoi solution : StringToIntegerAtoi.values()) {
                assertWithMessage(sample.getMessage())
                    .that(solution.myAtoi(sample.input.s))
                    .isEqualTo(sample.output);
            }
        }
    }

    public static class TypedSample extends Sample<TypedSample.Input, Integer> {
        public static class Input {
            public String s;
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/StringToIntegerAtoiTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class StringToIntegerAtoiTest : SampledTest() {
    @Test
    fun test() =
        StringToIntegerAtoi.entries.forEach { solution ->
            getSamples<Array<StringToIntegerAtoiSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.myAtoi(input.s))
                    .isEqualTo(output)
            }
        }
}
